### PR TITLE
One fix per day 15

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java [hunk 1]
```diff
@@ -369,22 +369,22 @@ public String consumeLiteral(Charset charset) throws DecodingException {
...
         } else {
-            try (FastByteArrayOutputStream out = new FastByteArrayOutputStream();
-                 InputStream in = consumeLiteral(false)) {
-                IOUtils.copy(in, out, 2048);
-                byte[] bytes = out.toByteArray();
-                ByteBuffer buffer = ByteBuffer.wrap(bytes);
+            ImmutablePair<Integer, InputStream> literal = consumeLiteral(false);
+            try (InputStream in = literal.right) {
+                Integer size = literal.left;
+                byte[] data = IOUtils.readFully(in, size);
```
Comment (chibenwa) [inline protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java:375]: I know this is orthogonal to your code change, but shouldn't we consider using a DeferredFileOutputStream here as the literal can be big and screw our heap?

File: server/container/util/src/main/java/org/apache/james/util/InputStreamUtils.java [hunk 2]
```diff
@@ -21,14 +21,14 @@
...
 import java.io.InputStream;
 import java.io.SequenceInputStream;
-import java.util.Arrays;
-import java.util.stream.Stream;
+import java.util.Collections;
+
+import com.google.common.collect.Lists;
 
 public class InputStreamUtils {
     public static InputStream concat(InputStream inputStream, InputStream... additionalInputStreams) {
```
Comment (chibenwa) [inline server/container/util/src/main/java/org/apache/james/util/InputStreamUtils.java:29]: This method is only used by tests where we could easily concat byte arrays instead.

Should we?

File: protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java (hunk 1 above)
Comment (mbaechler) [reply protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java:375]: We could. Some subclasses handle a size limit, some don't. I would not be a refactoring anymore (:

Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java:375]: I know. I'm just making a remark for future changes :-)

File: server/container/util/src/main/java/org/apache/james/util/InputStreamUtils.java (hunk 2 above)
Comment (mbaechler) [reply server/container/util/src/main/java/org/apache/james/util/InputStreamUtils.java:29]: I don't have anything against this class, we could add some features to centralize stream handling so I'd rather keep it

**Comment by mbaechler** [comment]
test this please

File: protocols/netty/src/main/java/org/apache/james/protocols/netty/NettyProtocolTransport.java [hunk 3]
```diff
@@ -126,26 +125,8 @@ protected void writeToClient(InputStream in, ProtocolSession session, boolean st
                     channel.write(new ChunkedStream(new ExceptionInputStream(e)));
                 }
                 return;
```
Comment (rouazana) [inline protocols/netty/src/main/java/org/apache/james/protocols/netty/NettyProtocolTransport.java:127]: what happens else?

Comment (mbaechler) [reply protocols/netty/src/main/java/org/apache/james/protocols/netty/NettyProtocolTransport.java:127]: see https://github.com/linagora/james-project/pull/3302/files/8b3db4a16b2b8c94fcb9c014d1a45099ae8d123d..f025913061013cd51172b1528aed794de326a0a2#diff-7a9220d5581cf0c1ae4275520c3091d0L149 it writes it another way

**Comment by chibenwa** [comment]
Merge

**Pull request closed without merging**